A36-xds-for-servers.md

```diff
@@ -299,36 +299,94 @@ reference" and "release a reference" on the shared instance, or similar
+create a server config fetcher for xDS, and register it with a server
+thereafter. The xDS server config fetcher will also create the `XdsClient`
+object needed to communicate with the control plane.
+
+```C
+typedef struct {
+  void (*on_serving_status_update)(void* user_data, const char* uri,
+                                   grpc_status_code code,
+                                   const char* error_message);
+  void* user_data;
+} grpc_server_xds_status_notifier;
+
+typedef struct grpc_server_config_fetcher grpc_server_config_fetcher;
+
+/** Creates an xDS config fetcher. */
+GRPCAPI grpc_server_config_fetcher* grpc_server_config_fetcher_xds_create(
+    grpc_server_xds_status_notifier notifier, const grpc_channel_args* args);
+
+/** Destroys a config fetcher. */
+GRPCAPI void grpc_server_config_fetcher_destroy(
+    grpc_server_config_fetcher* config_fetcher);
 
-C will need to expose an API for C++ to utilize. We expect there will be an xDS
-filter that will be injected into the server. But we do not focus on that here
-and leave that as an implementation detail to be part of a future gRFC. Since C
-is monolithic and we are fine with xDS using internal APIs, there is quite a bit
-more flexibility in design options available for C than Java and Go.
-
-Create an `XdsServerBuilder` that mirrors the `ServerBuilder` API. It may be
-possible to implement it via delegating to a `ServerBuilder` instance, but that
-is an implementation detail.
-
-The `XdsServerBuilder` will need to use a C core API (e.g., a channel arg to
-pass a plugin instance) to enable xDS support in the built `Server`. The
-`XdsServerBuilder` can indirectly create the `XdsClient` and plumb it to filters
-or other code that may need it. Notably, it will _not_ be passed to the
-`ServerCredential`; the `ServerCredential` will be passed the configuration to
-use for each connection, like `ChannelCredential`, so won’t need to use the
-`XdsClient` for watches on the ADS stream. Since many `Server` methods are not
-`virtual`, the `Server` will _not_ be wrapped to add xDS functionality. This
-means the C server will need to be responsible for shutting down the
-`XdsClient`.
-
-Since `ServerBuilder` has few virtual methods, the `XdsServerBuilder` will not
-be directly interchangeable. We believe users will be minimally impacted by
-needing to refer to the precise type, especially with the availability of
-templates. If this becomes a problem more methods could become `virtual`.
+/** Sets the server's config fetcher.  Takes ownership. Must be called before 
+    adding ports. */
+GRPCAPI void grpc_server_set_config_fetcher(
+    grpc_server* server, grpc_server_config_fetcher* config_fetcher);
+```
+
+The server needs to be configured with the config fetcher before we add ports to
+the server. Additionally, when using a server config fetcher, the bound port
+might not be available immediately as a return value of the
```

Comment:
I think we should say this a little more strongly: it means that it is not supported to use an ephemeral port with and xDS-enabled gRPC server.

Reply:
The reason for the not-so-strong language is to allow for flexibility to add support for wildcard ports, unless we want to say that we are never going to consider doing that.

Reply:
I think the current language is so fuzzy that it really doesn't tell the reader anything useful.  If the concern is that we want to be able to change this later, then let's just say something like "Note that we will initially not support ephemeral ports on an xDS-enabled server, but we might add that later."